main_test.go

```diff
@@ -8,15 +8,38 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestServeRoot(t *testing.T) {
-	req, err := http.NewRequest("GET", "/", nil)
-	assert.NoError(t, err)
+type test struct {
+	name            string
+	method          string
+	route           string
+	handler         http.HandlerFunc
+	statusCode      int    // expected status code
+	resBodyMustHave string // expected response body
+}
+
+// testing new routes and handlers requires only setting more test cases
+var cases = []test{
+	{"root", "GET", "/", serveRoot, http.StatusOK, "<!DOCTYPE html>"},
```

Comment:
these tests are important because getting the templates out of order (parent -> child) means that the request returns with `200 OK` but has no content.

here we can give it every route, it's handler, and a required body content to ensure every route is testable.